### ISSUE TITLE
docs: Algolia search needs to be optimized

### ISSUE THREAD

**Issue by mvdluit** [angular/angular#56035]
### Describe the problem that you experienced

When performing a search query on angular.dev, the results often do not show the pages with the highest relevancy for that term.

Example: Search for `Dependency Injection`. A page with high relevancy would be https://angular.dev/guide/di or https://angular.dev/guide/di/dependency-injection.

The results for this query are: 
- [Testing Services - Testing services with the TestBed](https://angular.dev/guide/testing/services#testing-services-with-the-testbed)
- [Interceptors - Configuring Interceptors](https://angular.dev/guide/http/interceptors#configuring-interceptors)
- [Common Routing Tasks - LocationStrategy and browser URL styles](https://angular.dev/guide/routing/common-router-tasks#locationstrategy-and-browser-url-styles)
- [DevTools overview - Visualize resolution paths](https://angular.dev/tools/devtools#visualize-resolution-paths)
- 4 reference results leading to pages about errors, the roadmap and ngModule explanation.

The search experience varies per query. Searching for `HttpClient` for example does show some results from the HttpClient guide, but those results will direct you to a very specific subsection of the guide, requiring you to navigate/search within the guide to find relevant general info. (e.g. first result for `HttpClient` is [Interceptors - Configuring Interceptors](https://angular.dev/guide/http/setup#withinterceptorsfromdi) where the overview page https://angular.dev/guide/http would be more relevant as I did not include anything about interceptors in my query).

I'm not familiar with Algolia, but I would expect that they offer a way to add a relevancy weighting to indexed pages for certain search terms.

### Describe what you want to experience that would fix the problem

I would hope to get relevant search results, which require minimal navigating through guides to find relevant info.

**Comment by anisabboud** [comment]
Algolia search indexing is broken since the initial release and hasn't been improved. See #53088

**Comment by mvdluit** [comment]
> This should be the top priority as the docs are absolutely unusable. I wanted to know more about `replaceUrl` option, I searched for:
> 
> * replaceUrl -> no results
> * router -> unrelated results
> * navigateByUrl -> unrelated results

I agree this needs be a priority. If you want information on a specific API (or in the above example a property on an Interface) it's often quicker to just go straight to the source code here on GitHub. Right now if you want to find `replaceUrl`, you would have to know that it's a property on `NavigationBehaviorOptions` which you can then look up in the [API reference](https://angular.dev/api). This is however **not a reasonable workaround** for people who are relatively new to Angular.

Compared to the search on https://v17.angular.io/docs, the ADEV search feels like a giant step back. Both in relevancy of results and amount of results returned for a query. Even if I search for a more general term, the search results on AIO allow me to go straight to a more specific search result.

@twerske I know that improving developer experience (and streamlining the on-boarding process for new developers) is a current goal mentioned on the roadmap. Could you please have a look at this, as I feel addressing/prioritizing this issue could lead to a huge improvement to the overall usability of the docs.

**Comment by aYorky** [comment]
Just to add my voice to the choir, the angular.dev search is just utterly unusable, and I dread having to open it every time I need to look something up. I was just looking for some information on the `Location` class, and angular.io immediately presents a search result that goes directly to that doc. Meanwhile angular.dev presents: Hierarchical injectors, Common Routing Tasks, Deploy multiple locales, security, etc. None of which are at all relevant, _let alone the the class with the exact name I searched for and is immediately relevant_. Thanks @mvdluit for the angular.io link, which I have now bookmarked. I wasn't sure how to get there anymore since going directly to "angular.io" now reroutes to angular.dev.

**Comment by mvdluit** [comment]
@aYorky You're welcome. Please note that the AIO docs will not be updated anymore, and that features introduced after v17 will not show up there. It will remain valid for some time for stable API's introduced before v18.

**Comment by aYorky** [comment]
@mvdluit Yeah I noticed it had that warning. I'm not upgrading to A18 any time soon and I'm not really using any of the "cutting edge" features of A17, so it'll work just fine for just about anything I need to reference.

**Comment by mvdluit** [comment]
The search results have significantly improved with https://github.com/angular/angular/pull/57281 and https://github.com/angular/angular/pull/57298

But it still seems that things like interface properties are not available in the search results. e.g. `replaceUrl` (property on `NavigationExtras`). I feel it would be very useful if these would also appear in the search results and link to the reference they belong to.

**Comment by JeanMeche** [comment]
This issue was also mentionned by #57744, it would be great if algolia also indexed methos/properties. 

@twerske, who has the hands on that ?

**Comment by JeanMeche** [comment]
Looks like the result have improved. 

Please open another issue if you have other specific search issues !
